bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c

```diff
@@ -157,30 +157,41 @@ Java_com_bugsnag_android_ndk_NativeBridge_deliverReportAtPath(
       jclass interface_class =
           bsg_safe_find_class(env, "com/bugsnag/android/NativeInterface");
       if (interface_class == NULL) {
+        pthread_mutex_unlock(&bsg_native_delivery_mutex);
         return;
       }
 
       // lookup NativeInterface.deliverReport()
       jmethodID jdeliver_method = bsg_safe_get_static_method_id(
           env, interface_class, "deliverReport", "([B[BLjava/lang/String;)V");
       if (jdeliver_method == NULL) {
+        pthread_mutex_unlock(&bsg_native_delivery_mutex);
```

Comment:
All of these mutex unlocks make me nervous. All it takes is a future modification that fails to cover this and we get deadlocks :/

Can we do a C-style RAII setup (i.e. `goto success` and `goto fail` which handle cleanup)

Reply:
Updated to use a RAII setup for cleanup.